### PR TITLE
Push Disciplina with runtime deps to Cachix

### PR DESCRIPTION
**Pull request by lukateras**
One can use cache by running:

```
nix-env -f '<nixpkgs>' -iA cachix
sudo cachix use disciplina
````

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -27,11 +27,12 @@ let
         "scripts/.*"
         "secrets/.*"
         "specs/.*"
+        ".*\\.nix"
```
Comment (yorickvP) [inline default.nix]: This should happen automatically, since these are not whitelisted

File: release.nix [hunk 2]
```diff
@@ -15,6 +15,20 @@ let
 in
 
 rec {
+  disciplina-haddock = with lib;
+    let
+      drvs = filterAttrs (const (hasAttr "doc")) project;
+      docs = mapAttrs (const (getOutput "doc")) drvs;
+      globs = map (doc: "${doc}/share/doc/*") (attrValues docs);
+    in
+    runCommand "disciplina-haddock" {} ''
```
Comment (yorickvP) [inline release.nix]: [pkgs.linkFarm](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/trivial-builders.nix#L146) "disciplina-haddock" ?

File: shell.nix [hunk 3]
```diff
@@ -5,6 +5,7 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [
     binutils
+    cachix
```
Comment (yorickvP) [inline shell.nix]: should this be a build input for `stack --nix`?

Comment (yegortimoshenko) [reply shell.nix]: See DisciplinaOU/disciplina-nixops@627e9ba7981b94e9b8bc63d8e96bda93be1fbc01.

File: release.nix (hunk 2 above)
Comment (yegortimoshenko) [reply release.nix]: Current approach looks much better in Buildkite artifacts output.

File: default.nix (hunk 1 above)
Comment (yegortimoshenko) [reply default.nix]: I'll check, but I think I added it because I had rebuilds on *.nix changes.

Comment (yegortimoshenko) [reply default.nix]: Nah, I removed it, immediately caused a rebuild.

**Pull request merged**